### PR TITLE
Audits `XrpEscrowFinish` Protobuf Wrapper

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

This PR adds additional logic validations and descriptive error throwing to the class `XrpEscrowFinish`.

### Context of Change

Another component of our attempt to make protobuf wrapper classes as safe and informative as possible. An SDK safety net.

### Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)

## Before / After

`XrpEscrowFinish` performs logic/error checking, based on the business logic in the docs.

## Test Plan

New tests added to confirm that errors are correctly thrown. Existing tests were edited to properly account for the new errors. CI passes.

## Future Tasks
This is part of a series, with all subtasks listed here: https://app.asana.com/0/1188384817180008/1193021408531788/f

### CODE REVIEW

File: src/XRP/model/xrp-escrow-finish.ts [hunk 1]
```diff
@@ -20,22 +21,34 @@ export default class XrpEscrowFinish {
...
-      return undefined
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'EscrowFinish protobuf is missing required `owner` field.',
+      )
     }
     const ownerXAddress = XrpUtils.encodeXAddress(
       owner,
       undefined,
       xrplNetwork == XrplNetwork.Test || xrplNetwork == XrplNetwork.Dev,
```
Comment (tedkalaw) [inline src/XRP/model/xrp-escrow-finish.ts:35]: if you wanted to make these `===` i wouldn't be pressed

**Review by tedkalaw — APPROVED** [review]
yeet

File: src/XRP/model/xrp-escrow-finish.ts (hunk 1 above)
Comment (amiecorso) [reply src/XRP/model/xrp-escrow-finish.ts]: lol maybe Mayukha can explain to me the nuances of "pressed" in the context of gen Z?

**Review by amiecorso — APPROVED** [review]
yeet indeed

File: src/XRP/model/xrp-escrow-finish.ts (hunk 1 above)
Comment (mvadari) [reply src/XRP/model/xrp-escrow-finish.ts]: Unfortunately I'm only about half gen Z so I don't qualify for the gen Z dictionary

**Pull request merged**